### ISSUE TITLE
feat: Auto-expand single-element folders

### ISSUE THREAD

**Issue by Winand** [ForkIssues/TrackerWin#2270]
Is it possible to collapse folders with one element in File Tree like VS Code does? Or at least automatically expand the tree up to `appName` folder when I open `scala` folder. Often I need to open the Blame window for a specific file and it takes too much time to expand every single folder in the tree.
![image](https://github.com/user-attachments/assets/ad00a5a1-acd4-48fb-907e-3ce7cbd63a3c)

**Comment by DanPristupov** [comment]
> Is it possible to collapse folders with one element in File Tree like VS Code does?

No. That would make opening a history (or any other context menu operation) for a folder in the middle (like cc) difficult.

> Or at least automatically expand the tree up to appName folder when I open scala folder.

We'll think about this.

> Often I need to open the Blame window for a specific file and it takes too much time to expand every single folder in the tree.

`Ctrl+P → Blame → <filename>`
